perceval/converters/abstract_converter.py

```diff
@@ -49,10 +49,14 @@ class AGateConverter(ABC):
     Converter class for gate based Circuits to perceval processor
     """
 
-    def __init__(self, backend_name: str = "SLOS", source: Source = Source()):
+    def __init__(self, backend_name: str = "SLOS", source: Source = None, noise_model: NoiseModel = NoiseModel()):
         self._converted_processor = None
         self._input_list = None  # input state in list
-        self._noise_model = NoiseModel()
+        if source is not None:
```

Comment:
Ok, but here `self._source` is never used.

I'd suggest:
```python
self._noise_model = NoiseModel()
if source is not None:
    get_logger().warn('DeprecationWarning: Call with deprecated argument "source", '
                              'please use "noise_model=NoiseModel()" instead')
    self._noise_model = NoiseModel(transmittance=1-source._losses,
                                   brightness=source._emission_probability,
                                   g2=source._multiphoton_component,
                                   indistinguishability=source._indistinguishability,
                                   g2_distinguishable=(source._multiphoton_model=='distinguishable')
                                   )
if noise_model is not None:
    self._noise_model = noise_model
```
And set the default value of `noise_model` to `None` too

Reply:
You are absolutely correct, i completely forgot that my update would render the source unused. Fixed.